gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -0,0 +1,829 @@
+        auto column = columns->Get(i);
+        auto name = column->name()->c_str();
+        auto type = toOGRFieldType(column->type());
+        OGRFieldDefn field(name, type);
+        m_poFeatureDefn->AddFieldDefn(&field);
+    }
+}
+
+OGRFlatGeobufLayer::~OGRFlatGeobufLayer()
+{
+    if (m_create) {
+        CPLDebug("FlatGeobuf", "Request to create %zu features", m_featuresCount);
+        size_t c;
+        uint64_t offset = 0;
+
+        //const char *filename = CPLFormFilename("", m_pszLayerName, "fgb");
+        CPLDebug("FlatGeobuf", "m_pszFilename: %s", m_pszFilename);
+        VSILFILE *fp = VSIFOpenL(m_pszFilename, "wb");
+
+        if(fp == nullptr) {
+            CPLError(CE_Fatal, CPLE_OpenFailed,
+                        "Failed to create %s:\n%s",
+                        m_pszFilename, VSIStrerror(errno));
+            return;
+        }
+
+        c = VSIFWriteL(&magicbytes, sizeof(magicbytes), 1, fp);
+        CPLDebug("FlatGeobuf", "Wrote magicbytes (%zu bytes)", c * sizeof(magicbytes));
+        offset += c;
+
+        Rect extent = calcExtent(m_featureItems);
+        const auto extentVector = extent.toVector();
+
+        FlatBufferBuilder fbb;
+        auto columns = writeColumns(fbb);
+
+        uint16_t indexNodeSize = bCreateSpatialIndexAtClose ? 16 : 0;
+
+        flatbuffers::Offset<Crs> crs = 0;
+        if (m_poSRS) {
+            int nAuthorityCode = 0;
+            const char* pszAuthorityName = m_poSRS->GetAuthorityName( nullptr );
+            if ( pszAuthorityName == nullptr || strlen(pszAuthorityName) == 0 )
+            {
+                // Try to force identify an EPSG code.
+                CPLDebug("FlatGeobuf", "AutoIdentifyEPSG");
+                m_poSRS->AutoIdentifyEPSG();
+
+                pszAuthorityName = m_poSRS->GetAuthorityName(nullptr);
+                if (pszAuthorityName != nullptr && EQUAL(pszAuthorityName, "EPSG"))
+                {
+                    const char* pszAuthorityCode = m_poSRS->GetAuthorityCode(nullptr);
+                    if ( pszAuthorityCode != nullptr && strlen(pszAuthorityCode) > 0 )
+                    {
+                        /* Import 'clean' SRS */
+                        m_poSRS->importFromEPSG( atoi(pszAuthorityCode) );
+
+                        pszAuthorityName = m_poSRS->GetAuthorityName(nullptr);
+                    }
+                }
+            }
+            if ( pszAuthorityName != nullptr && strlen(pszAuthorityName) > 0 )
+            {
+                // For the root authority name 'EPSG', the authority code
+                // should always be integral
+                nAuthorityCode = atoi( m_poSRS->GetAuthorityCode(nullptr) );
+            }
+
+            // Translate SRS to WKT.
+            char *pszWKT1 = nullptr;
+            char *pszWKT2 = nullptr;
+            const char* const apszOptionsWkt1[] = { "FORMAT=WKT1_GDAL", nullptr };
+            const char* const apszOptionsWkt2[] = { "FORMAT=WKT2_2015", nullptr };
```

Comment:
any reason not to export to FORMAT=WKT2_2018 systemetically ? This is a new format. No backward compatibility issue

Reply:
No reason other than I thought it still was a work in progress version and that it's naming/versioning/compatiblity might be uncertain. I wonder if it makes more sense to name the properties in my format to simply `wkt1`, `wkt2_2015` and `wkt2_2018` then instead of basing them on GeoPackage `definition_*` names.

Reply:
WKT2_2018 (will probably be named 2019 but whatever) is in the publishing/editorial stages at ISO and similarly at OGC, so its content is finalized now.

Reply:
Right ok, compatibility story not entirely clear - incompatible with wkt 2015 so should have dedicated property in any supporting format?

Reply:
It depends which readers you want to address. If this is only GDAL, WKT2_2018 would be enough. Having support for 3 different WKT variants seems overkill

Reply:
I want flatgeobuf (and as such any crs meta) to be easily consumed in as many places as possible. Of course, I suppose most use cases will simply involve the authority code. And I suppose the latest WKT can be translated into an older version by a capable parser i.e GDAL/proj for those who need it.. so perhaps it's fine to have WKT 2018 as the baseline for a (possibly) emerging format. But I'm fumbling in the dark to cater for a niche use case that I'm not very familiar with (very seldom work with data that has custom crs).

Reply:
Thinking about this again I'm leaning that it would be best to leave out version completely and just store the definition as a single `wkt` string with unspecified version indicating that an interested consumer must introspect it themselves i.e using the instructions at B.8 of 12-063r5.